src/main/java/org/mule/extension/aggregator/internal/privileged/executor/AbstractAggregatorExecutor.java

```diff
@@ -213,6 +222,30 @@ public void start() throws MuleException {
     }
   }
 
+  private void executeDeferActions() {
+    quorumManager = new Thread(() -> {
+      boolean startCompleted = false;
+      while (!startCompleted) {
+        try {
+          Thread.sleep(200);
+          upgradeAggregatedContentIfNeeded();
+          setRegisteredAsyncAggregationsAsNotScheduled();
+          startCompleted = true;
+        } catch (Exception e) {
+          if (e.getClass().getName().contains(QUORUM_EXCEPTION)) {
+            LOGGER.warn("Quorum constraint not satisfied");
+            if (LOGGER.isDebugEnabled())
+              LOGGER.debug(e.getMessage());
+          } else {
+            LOGGER.error(e.getMessage(), e);
+            Thread.interrupted();
```

Comment:
I don't think this  is ok. 

Here you are asserting for the thread's interrupted flag and then clearing it. If this thread was interrupted, you definitely don't want to clear the flag, otherwise the task could keep running forever. 

You should add extra logic to stop the task if the thread was interrupted and return.

Reply:
We remove that logic, it is handled through the ScheduleService